nixos/modules/services/x11/display-managers/lightdm.nix

```diff
@@ -72,17 +40,17 @@ let
       ${cfg.extraSeatDefaults}
     '';
 
-  gtkGreeterConf = writeText "lightdm-gtk-greeter.conf"
-    ''
-    [greeter]
-    theme-name = Adwaita
-    icon-theme-name = Adwaita
-    background = ${cfg.background}
-    '';
-
 in
 {
+  # Note: the order in which lightdm greeter modules are imported
+  # here determines the default: later modules (if enable) are
+  # preferred.
```

Comment:
Indeed, we can check and test the behavior properly when we have another greeter implementation.

However, could we split the single option, that holds the expected configuration, into two separate options; greeter.package & greeter.name. That way we can add proper types and explanation to the config options, and from what I understand of the nix syntax, shouldn't break backwards compatibility.